python/pyspark/tests/test_pin_thread.py

```diff
@@ -0,0 +1,156 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import time
+import random
+import threading
+import unittest
+
+from pyspark import SparkContext, SparkConf
+
+
+class PinThreadTests(unittest.TestCase):
+    # These tests are in a separate class because it uses
+    # 'PYSPARK_PIN_THREAD' environment variable to test thread pin feature.
```

Comment:
just a general question about pyspark code -- how come we don't use docstrings for these class & method comments?  Seems we do in a few cases, but not consistently, mostly they're just comments.

Reply:
Yeah, agree. I don't think there's kind of a consistent rule for that. I usually just write comments when the contents are not directly related to the description of the class or method.

Actually I don't bother docstrings often in test classes because it's unlikely the docstrings in class are read via pydoc (e.g., `help(PinThreadTests)` in Python shall) or generated as a doc.